packages/jsii-rosetta/lib/snippet.ts

```diff
@@ -66,7 +66,7 @@ export function completeSource(snippet: TypeScriptSnippet) {
 function parametersFromSourceDirectives(
   source: string,
 ): [string, Record<string, string>] {
-  const [firstLine, rest] = source.split('\n', 2);
+  const [firstLine, ...rest] = source.split('\n');
```

Comment:
Note - there was a bug right here that my test highlighted (tests with only 1 line of snippet have missed this entirely).

The culprit being that JavaScript's `String.split` function doesn't work like most folks expect it to when it comes to the limit: instead of limiting the amount of cuts that will be made in the source (like in java) and having the last slice be "the rest", JavaScript cuts on all separators, and the limit restricts the amount of slices returned.

In other words:
```java
// Example.java
final String abc = "a,b,c";
abc.split(",", 2);
//=> ["a", "b,c"]
```

Versus
```js
// Example.js
const abc = 'a,b,c';
abc.split(',', 2);
//=> ['a', 'b'] // Omitting slice #3, 'c'.
```